pytket/tests/routing_test.py

```diff
@@ -1,876 +0,0 @@
-# Copyright 2019-2022 Cambridge Quantum Computing
```

Comment:
I've now added back a large number of tests (over the python and c++ code bases) and am now confident that we capture the previous test coverage + more. Similarly to pytket.architecture, pytket.placement etc I have moved the tests into independent files, instead of under the "routing" banner.